app/server/lib/dbUtils.ts

```diff
@@ -53,54 +53,65 @@ export function getConnectionName() {
  * Get a connection to db if one exists, or create one. Serialized to
  * avoid duplication.
  */
+let gristDataSource: DataSource | null = null;
 const connectionMutex = new Mutex();
-
-export async function getOrCreateConnection(): Promise<Connection> {
+export async function getOrCreateConnection(): Promise<DataSource> {
   return connectionMutex.runExclusive(async() => {
-    try {
-      // If multiple servers are started within the same process, we
-      // share the database connection.  This saves locking trouble
-      // with Sqlite.
-      const connection = getConnection();
-      return connection;
-    } catch (e) {
-      if (!String(e).match(/ConnectionNotFoundError/)) {
-        throw e;
+
+    // If multiple servers are started within the same process, we
+    // share the database connection.  This saves locking trouble
+    // with Sqlite.
+    if (!gristDataSource?.isInitialized) {
+      let settings = getTypeORMSettings();
+      if (settings.type === 'postgres') {
+        // We're having problems with the Postgres JIT compiler slowing
+        // down a particular query, so we'll turn it off for this
+        // session.
+        //
+        // If some day Postgres's JIT compiler gets smarter and has a
+        // better cost function that knows it's a bad idea to compile
+        // certain queries, we might then want to revisit this
+        // workaround and remove it.
+        settings = getTypeORMSettings({ extra: { options: "-c jit=off" } });
```

Comment:
This looks great! Could you add a link to the relevant postgres documentation? The `-c jit=off` is getting supplied at the connection level?

Reply:
Yes, the `-c jit=off` option is supplied at the connexion level.

I've added a link to the relevant Postgres documentation.

Reply:
So the thing I'm having trouble finding myself is the documentation on how that option is presented. For example, as a reviewer, if the `-c` were supposed to be `-C` I should catch that, but I'm failing at finding where that is specified.

Reply:
Oh!

They're command-line options to `libpq`:

https://www.postgresql.org/docs/17/config-setting.html#CONFIG-SETTING-SHELL

I'll add this link.

Reply:
To be crystal clear: TypeORM doesn't start a Postgres server, so these `-c` parameters are not being given to the server. I confirmed, just in case, that after connecting Grist to Postgres, other connexions to the same Postgres server still have JIT compilation enabled.

Reply:
Thanks @jordigh, that's what I was looking for, especially the `For historical reasons, the format of PGOPTIONS is similar to that used when launching the postgres command` that explains why the options look the way they do.

Reply:
Hello ! This breaks our prod deployment as `TYPEORM_EXTRA` in which we define an `extra` key already gets overwritten (the extra keys are not merged)

Will provide a fix in a different PR